packages/backend/src/flows/flow-version/flow-version.service.ts

```diff
@@ -159,24 +161,45 @@ function validateTrigger(settings: PieceTriggerSettings) {
   return validateProps(trigger.props, settings.input);
 }
 
-// TODO replace with proper validation, currently it's only validate wether the input is there or not, It should check schema and types as well.
-function validateProps(props: Record<string, PieceProperty>, input: Record<string, unknown>) {
+function validateProps(props: PieceProperty, input: Record<string, unknown>) {
+  const propsSchema = buildSchema(props);
+  const propsValidator = TypeCompiler.Compile(propsSchema);
+  return propsValidator.Check(input);
+}
+
+function buildSchema(props: PieceProperty): TSchema {
   const entries = Object.entries(props);
+  let propsSchema: Record<string, TSchema> = {};
   for (let i = 0; i < entries.length; ++i) {
-    const property: PieceProperty = entries[i][1];
+    const property = entries[i][1];
     const name: string = entries[i][0];
-    if (property.required && isMissing(input[name])) {
-      return false;
+    switch (property.type) {
+      case PropertyType.SHORT_TEXT:
+      case PropertyType.LONG_TEXT:
+        propsSchema[name] = Type.String({
+          minLength: property.required ? 1 : undefined
+        });
+        break;
+      case PropertyType.CHECKBOX:
+        propsSchema[name] = Type.Boolean({});
+        break;
+      case PropertyType.NUMBER:
+        propsSchema[name] = Type.Number({});
+        break;
+      case PropertyType.DROPDOWN:
+        propsSchema[name] = Type.Any({});
+        break;
+      case PropertyType.OAUTH2:
+        // Only accepts connections variable.
+        propsSchema[name] = Type.RegEx(RegExp('\$\{connections.(.*?)\}', 'g'));
+        break;
+    }
```

Comment:
There's `SECRET_TEXT`, and `CUSTOM_AUTH` now?

Reply:
It doesn't matter, The frontend should always input the reference `${connections.NAME}`